src/gsy_e/gsy_e_core/sim_results/endpoint_buffer.py

```diff
@@ -195,14 +196,35 @@ def _create_results_handler(should_export_plots):
     def create_hierarchy_stats(self, area: "AreaBase"):
         """Calculate hierarchy related statistics."""
         hierarchy_self_consumption_list = {}
-        self._calc_hierarchy_stats(area, 0, hierarchy_self_consumption_list)
+        hierarchy_self_consumption = {}
+        self._calc_hierarchy_self_consumption(area, 0, hierarchy_self_consumption_list)
         for level, consumption_list in hierarchy_self_consumption_list.items():
             if len(consumption_list):
-                self.hierarchy_self_consumption[level] = sum(consumption_list) / len(
-                    consumption_list
-                )
+                hierarchy_self_consumption[level] = sum(consumption_list) / len(consumption_list)
+        if (
+            len(hierarchy_self_consumption) > 0
+            and hierarchy_self_consumption[next(iter(hierarchy_self_consumption))] == 0
+        ):
+            # early return in case there is no self consumpton at all (in order to not devide by 0)
+            return
 
-    def _calc_hierarchy_stats(self, area: "AreaBase", level: int, results: Dict):
+        for level, self_consumption in hierarchy_self_consumption.items():
+            if level + 1 not in hierarchy_self_consumption:
+                # lowest level case:
+                self.hierarchy_self_consumption_percent[level] = (
+                    self_consumption / hierarchy_self_consumption[0]
+                ) * 100
+            else:
+                self.hierarchy_self_consumption_percent[level] = (
+                    (self_consumption - hierarchy_self_consumption[level + 1])
+                    / hierarchy_self_consumption[0]
+                ) * 100
+        if self.hierarchy_self_consumption_percent:
+            assert isclose(
+                sum(self.hierarchy_self_consumption_percent.values()), 100
+            ), "Self consumption percentages do not sum up to 100%"
```

Comment:
This is a great assert, it will always confirm that this feature is working as expected, thanks!